### PR TITLE
[[ KeyIsDown ]] Implement key is down

### PR DESCRIPTION
**Pull request by montegoulding**
This patch implements modifier key state access for LCB.

### CODE REVIEW

File: engine/src/engine.lcb [hunk 1]
```diff
@@ -634,4 +636,30 @@ begin
...
 end syntax
 
+/**
+Summary:    Returns true if the key is down
+
+Returns:    The state of the key
+
+Description:
+Use 'the ... key is down' to determine if the key was down at the start of the
+current evend. Use 'the ... key is currently down' to determine if the key is
```
Comment (livecodeali) [inline engine/src/engine.lcb]: `evend`

File: engine/src/engine.lcb [hunk 2]
```diff
@@ -634,4 +636,30 @@ begin
...
+down at the time it is being checked.
+
+As in script command and control keys return the state of the same key on non-macOS
```
Comment (livecodeali) [inline engine/src/engine.lcb]: can you put a comma after `script` here

Comment (montegoulding) [reply engine/src/engine.lcb]: done

File: engine/src/engine.lcb (hunk 1 above)
Comment (montegoulding) [reply engine/src/engine.lcb]: done

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 8f6d018

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 8f6d0189f0dde54cd570ad4a81a1e4ce242acda1

**Comment by livecode-vulcan** [comment]
😎 test success 8f6d0189f0dde54cd570ad4a81a1e4ce242acda1

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/64)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/56)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/56)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/56)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/56)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1578)
* try-community-universal-ios-iphoneos11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.4/builds/105)
* try-community-universal-ios-iphonesimulator11.4: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.4/builds/105)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1608)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1602)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1598)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1635)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1575)

**Pull request merged**